frame/include/bli_type_defs.h

```diff
@@ -68,8 +70,8 @@ typedef          uint32_t guint_t;
 typedef           int64_t  gint_t;
 typedef          uint64_t guint_t;
 #else
-typedef   signed long int  gint_t;
-typedef unsigned long int guint_t;
+typedef         ptrdiff_t  gint_t;
+typedef            size_t guint_t;
```

Comment:
I guess you could use `intptr_t` and `uintptr_t` here, and remove the need for `stddef.h`? Otherwise looks good to me.